examples/api/demo_affine_image.py

```diff
@@ -21,25 +20,37 @@ def get_image():
     return Z
 
 
-if 1:
+def do_plot(ax, Z, transform):
+    im = ax.imshow(Z, interpolation='none',
+                   origin='lower',
+                   extent=[-2, 4, -3, 2], clip_on=True)
+
+    trans_data = transform + ax.transData
+    im.set_transform(trans_data)
+
+    # display intended extent of the image
+    x1, x2, y1, y2 = im.get_extent()
+    ax.plot([x1, x2, x2, x1, x1], [y1, y1, y2, y2, y1], "y--",
+            transform=trans_data)
+    ax.set_xlim(-5, 5)
+    ax.set_ylim(-4, 4)
 
-    # image rotation
 
-    fig, ax1 = plt.subplots(1, 1)
+if 1:
```

Comment:
Amended commits to remove this test and rebased onto latest master as I had to force-push anyway.